java/ActigraphReader.java

```diff
@@ -72,31 +78,41 @@ public static void readG3TXEpochs(
 
             // underscored are unused for now
             double sampleFreq = -1, accelerationScale = -1, _AccelerationMin, _AccelerationMax;
-            long _LastSampleTime, firstSampleTime=-1;
+            long startDate = -1, stopDate = -1, firstSampleTime=-1;
             String serialNumber = "";
+            String infoTimeZone = "01:00:00"; // default to be UTC time difference
 
             while (infoReader.ready()) {
                 String line = infoReader.readLine();
                 if (line!=null){
-                    String[] tokens=line.split(":");
+                    String[] tokens=line.split(": ");
                     if ((tokens !=null)  && (tokens.length==2)){
-                        if (tokens[0].trim().equals("Sample Rate")){
+                        if (tokens[0].trim().equals("Sample Rate"))
                             sampleFreq=Integer.parseInt(tokens[1].trim());
-                        } else if (tokens[0].trim().equals("Last Sample Time"))
-                            _LastSampleTime=GT3XfromTickToMillisecond(Long.parseLong(tokens[1].trim()));
+                        else if (tokens[0].trim().equals("First Sample Date"))
+                            firstSampleTime=GT3XfromTickToMillisecond(Long.parseLong(tokens[1].trim()));
                         else if (tokens[0].trim().equals("Acceleration Scale"))
                             accelerationScale=Double.parseDouble(tokens[1].trim());
                         else if (tokens[0].trim().equals("Acceleration Min"))
                             _AccelerationMin=Double.parseDouble(tokens[1].trim());
                         else if (tokens[0].trim().equals("Acceleration Max"))
                             _AccelerationMax=Double.parseDouble(tokens[1].trim());
                         else if (tokens[0].trim().equals("Start Date"))
-                            firstSampleTime=GT3XfromTickToMillisecond(Long.parseLong(tokens[1].trim()));
+                            startDate=GT3XfromTickToMillisecond(Long.parseLong(tokens[1].trim()));
+                        else if (tokens[0].trim().equals("Stop Date"))
+                            stopDate=GT3XfromTickToMillisecond(Long.parseLong(tokens[1].trim()));
                         else if (tokens[0].trim().equals("Serial Number"))
                             serialNumber=tokens[1].trim();
+                        else if (tokens[0].trim().equals("TimeZone"))
+                            infoTimeZone=tokens[1].trim();
                     }
                 }
             }
+
+            System.out.println("Timezone: " + infoTimeZone);
+            System.out.println("Start date (local UNIX): " + startDate);
+            System.out.println("Stop date (local UNIX): " + stopDate);
```

Comment:
This will print the pseudo/local UNIX times, correct? Is this here just for debugging?

Reply:
Not just for debugging but to print out what's being extracted from the meta file.